### PR TITLE
GTFS arrivals

### PR DESCRIPTION
**Pull request by ianweller**
OK, so there's a lot going on here.

I strongly recommend reading this pull request commit-by-commit in order to understand the development process and reasoning behind everything.

This should become a stable API for arrivals (i.e. these attributes should in theory not change), but I'm hesitant to call it that until we start building apps around real data as part of our senior design project.

tl;dr: :bus: :bus:

### CODE REVIEW

**Comment by ianweller** [comment]
A couple more notes:
- because the GTFS sample feed doesn't have parent/child stops, we lost test coverage for `busbus.entity.LazyEntityProperty`. I'm going to create a test case using a mocked GTFS feed after I get some work on `busbus.web` done.
- I promise to never create a pull request this size again.

File: busbus/provider/gtfs.py [hunk 1]
```diff
@@ -47,23 +72,136 @@ def __init__(self, provider, **data):
...
+
+class GTFSTrip(busbus.entity.BaseEntity):
+    __attrs__ = ('id', 'route', 'service', 'headsign', 'short_name', 'block',
+                 'shape', 'accessible', 'bikes_ok')
+
+    def __init__(self, provider, **data):
+        data['route'] = provider.get(busbus.Route, data['_route_id'])
+        data['service'] = provider.get(GTFSService, data['_service_id'])
+        if '_block_id' in data:
+            pass  # FIXME
```
Comment (asmallteapot) [inline busbus/provider/gtfs.py]: There are a few FIXMEs in this vicinity. The underlying data types may be out of scope for this PR, though.

File: busbus/provider/gtfs.py [hunk 2]
```diff
@@ -47,23 +72,136 @@ def __init__(self, provider, **data):
...
+        if 'exact_times' in data:
+            data['exact_times'] = {u'0': False, u'1': True}.get(
+                data['exact_times'], True)
+        provider._add_relation(GTFSTrip, data['_trip_id'], 'stop_times', self)
+        provider._add_relation(busbus.Stop, data['_stop_id'], 'trips',
+                               data['trip'])
+        super(GTFSStopTime, self).__init__(provider, **data)
+
+
+GTFS_FILENAME_MAP = OrderedDict([
```
Comment (asmallteapot) [inline busbus/provider/gtfs.py]: This large ordered dictionary is good for having the entire key mapping in one place. An alternate approach based around reading `__attrs__` and, optionally, designated method(s) for remapping GTFS column names to Python property names, could reduce the "big blob of data" factor here.

File: busbus/provider/gtfs.py (hunk 1 above)
Comment (ianweller) [inline busbus/provider/gtfs.py]: Yep, indeed out of scope for this PR, and something I'm hoping to fix soon.

**Comment by alexdgus** [comment]
Looks good.

**Comment by nicksaccount4tw** [comment]
I agree, looks good.

**Pull request merged**